x/incentives/keeper/distribute_test.go

```diff
@@ -92,6 +95,142 @@ func (suite *KeeperTestSuite) TestDistribute() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestDistributeToConcentratedLiquidityPools() {
+	fiveKRewardCoins := sdk.NewInt64Coin(defaultRewardDenom, 5000)
+	fifteenKRewardCoins := sdk.NewInt64Coin(defaultRewardDenom, 15000)
+
+	coinsToMint := sdk.NewCoins(sdk.NewCoin(defaultRewardDenom, sdk.NewInt(1000000)))
+	defaultGaugeStartTime := suite.Ctx.BlockTime()
+
+	incentivesParams := suite.App.IncentivesKeeper.GetParams(suite.Ctx).DistrEpochIdentifier
+	currentEpoch := suite.App.EpochsKeeper.GetEpochInfo(suite.Ctx, incentivesParams)
+
+	tests := map[string]struct {
+		// setup
+		numPools           int
+		tokensToAddToGauge sdk.Coins
+		gaugeStartTime     time.Time
+		poolType           poolmanagertypes.PoolType
+
+		// expected
+		expectErr             bool
+		expectedDistributions sdk.Coins
+	}{
+		"valid case: one poolId and gaugeId": {
+			numPools:              1,
+			gaugeStartTime:        defaultGaugeStartTime,
+			expectedDistributions: sdk.NewCoins(fiveKRewardCoins),
+			expectErr:             false,
+		},
+
+		"valid case: multiple gaugeId and poolId": {
+			numPools:              3,
+			gaugeStartTime:        defaultGaugeStartTime,
+			expectedDistributions: sdk.NewCoins(fifteenKRewardCoins),
+			expectErr:             false,
+		},
+		"invalid case: attempt to createIncentiveRecord with starttime < currentBlockTime": {
+			numPools:       1,
+			poolType:       poolmanagertypes.Concentrated,
+			gaugeStartTime: defaultGaugeStartTime.Add(-1 * time.Hour),
+			expectErr:      true,
+		},
+	}
+
+	for name, tc := range tests {
+		suite.Run(name, func() {
+			// setup test
+			suite.SetupTest()
+			// We fix blocktime to ensure tests are deterministic
+			suite.Ctx = suite.Ctx.WithBlockTime(defaultGaugeStartTime)
+
+			var gauges []types.Gauge
+
+			// prepare the minting account
+			addr := sdk.AccAddress([]byte("Gauge_Creation_Addr_"))
+			// mints coins so supply exists on chain
+			suite.FundAcc(addr, coinsToMint)
+
+			// make sure the module has enough funds
+			suite.App.BankKeeper.SendCoinsFromAccountToModule(suite.Ctx, addr, types.ModuleName, coinsToMint)
+
+			// prepare a CL Pool that creates gauge at the end of createPool
+			for i := 0; i < tc.numPools; i++ {
+				poolId := suite.PrepareConcentratedPool().GetId()
+
+				// get the gaugeId corresponding to the CL pool
+				gaugeId, err := suite.App.PoolIncentivesKeeper.GetPoolGaugeId(suite.Ctx, poolId, currentEpoch.Duration)
+				suite.Require().NoError(err)
+
+				// get the gauge from the gaudeId
+				gauge, err := suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, gaugeId)
+				suite.Require().NoError(err)
+
+				gauge.Coins = sdk.NewCoins(sdk.NewCoin(defaultRewardDenom, sdk.NewInt(5000)))
```

Comment:
Do we have a case where a gauge contains multiple coins? This should be added before merge IMO